Nested tuple destructuring in macros

```jl
m = Model()
@variable(m, x)
d = Dict((1,2) => 3)
@constraint(m, x == sum(i+j+k for ((i,j),k) in d))
```
```
ERROR: AssertionError: x.head == :escape
```
whereas
```
d = Dict((1,2) => 3)
sum(i+j+k for ((i,j),k) in d)
```
```
6
```

--- a/src/macros.jl
+++ b/src/macros.jl
@@ -206,7 +206,7 @@ localvar(x::Symbol) = _localvar(x)
 localvar(x::Expr) = Expr(:block, _localvar(x)...)
 _localvar(x::Symbol) = :(local $(esc(x)))
 function _localvar(x::Expr)
-    @assert x.head == :escape
+    @assert x.head in (:escape,:tuple)
     args = Any[]
     for t in x.args
         if isa(t, Symbol)

--- a/test/macros.jl
+++ b/test/macros.jl
@@ -806,4 +806,13 @@ end
         @test string(JuMP.constructconstraint!(x, :(<=))) == "x $leq 0"
         @test string(JuMP.constructconstraint!(x, :(==))) == "x $eq 0"
     end
+
+    @testset "Nested tuple destructuring" begin
+    m = Model()
+    @variable(m, x)
+    d = Dict((1,2) => 3)
+    @constraint(m, x == sum(i+j+k for ((i,j),k) in d))
+    @test m.linconstr[1].lb == m.linconstr[1].ub == 6
+    @test m.linconstr[1].terms == 1x
+    end
 end

